### ISSUE TITLE
[3.0.0pre1] NPE in mzta-menus.js

### ISSUE THREAD

**Issue by pradeepachan** [micz/ThunderAI#206]
### The problem

NPE while testing the tags functionality

### Which Operative System are you using?

Ubuntu

### Which version of Thunderbird are you using?

128.5.2esr (64-bit)

### Which version of ThunderAI has the issue?

128.5.2esr (64-bit)

### Which integration are you using?

Ollama API

### Additional information

This line:                        browser.tabs.sendMessage(tabs[0].id, {command: "getTags", tags: mail_tags, messageId: curr_message.id}); was giving error having curr_message as null.

Moved the below code before the if statement, instead of being within else. 

 

```
let curr_messages = null;
            switch(tabs[0].type){
                case 'mail':
                    curr_messages = await browser.mailTabs.getSelectedMessages();
                    curr_message = curr_messages.messages[0];
                    break;
                case 'messageDisplay':
                    curr_messages = await messenger.messageDisplay.getDisplayedMessage(tabs[0].id);
                    curr_message = curr_messages;
                    break;
                case 'messageCompose':
                    curr_messages = await browser.compose.getComposeDetails(tabs[0].id);
                    curr_message = curr_messages;
                    break;
            }
```

The relevant if clause is:

`if(!placeholdersUtils.hasPlaceholder(curr_prompt.text)){`

**Comment by micz** [comment]
Thank you for your submission.
May you tell me what you were doing when using ThunderAI? You were in the mails list or viewing an email in a single separate window?

**Comment by yashairavantchi** [comment]
I came across the same issue, when I click AI-->0. Add tags to this email I get the same `Error executing action for menu item prompt_add_tags: TypeError: curr_message is null` regardless of whether it's viewing in the mail list or separated in a single window.

**Comment by micz** [comment]
@yashairavantchi may you activate the debug in the addon options page and post here the lines from the console log?
Thank you.

**Comment by micz** [comment]
Have you changed the default add_tags prompt?

**Comment by yashairavantchi** [comment]
I have not modified the default add_tags prompt. To make sure, I removed the ThunderAI addon (with deleting all config files), re-added the addon, and restarted Thunderbird, but the 0. add tags option has disappeared, so I can't get back to that error:

![image](https://github.com/user-attachments/assets/196f8d35-d2ea-485b-b508-e777430bbfcc)

The version is definitely the correct version:
![image](https://github.com/user-attachments/assets/cdf2c969-f6fa-4bbf-b537-8d334ab19b5b)

**Comment by yashairavantchi** [comment]
Ok, I got the option to come back after enabling and disabling tags checkbox. Here is the console output:
```
[ThunderAI Logger | mzta-popup] Preparing data to load the popup menu: {"lastShortcutTabId":1,"lastShortcutTabType":"mail","lastShortcutPromptsData":[{"id":"prompt_add_tags","label":"Add tags to this email","type":"1"},{"id":"prompt_classify","label":"Classify","type":"0"},{"id":"prompt_reply","label":"Reply to this email","type":"1"},{"id":"prompt_reply_advanced","label":"Reply to this thread","type":"1"},{"id":"prompt_rewrite_formal","label":"Rewrite formal","type":"2"},{"id":"prompt_rewrite_polite","label":"Rewrite polite","type":"2"},{"id":"prompt_summarize_this","label":"Summarize this","type":"0"},{"id":"prompt_this","label":"Prompt this","type":"2"},{"id":"prompt_translate_this","label":"Translate this","type":"0"}],"lastShortcutFiltering":1} [mzta-logger.js:35:44](moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/mzta-logger.js)
[ThunderAI Logger | mzta-popup] _prompts_data: [{"id":"prompt_add_tags","label":"Add tags to this email","type":"1"},{"id":"prompt_classify","label":"Classify","type":"0"},{"id":"prompt_reply","label":"Reply to this email","type":"1"},{"id":"prompt_reply_advanced","label":"Reply to this thread","type":"1"},{"id":"prompt_rewrite_formal","label":"Rewrite formal","type":"2"},{"id":"prompt_rewrite_polite","label":"Rewrite polite","type":"2"},{"id":"prompt_summarize_this","label":"Summarize this","type":"0"},{"id":"prompt_this","label":"Prompt this","type":"2"},{"id":"prompt_translate_this","label":"Translate this","type":"0"}] [mzta-logger.js:35:44](moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/mzta-logger.js)
[ThunderAI Logger | mzta-popup] active_prompts: [{"id":"prompt_add_tags","label":"Add tags to this email","type":"1"},{"id":"prompt_classify","label":"Classify","type":"0"},{"id":"prompt_reply","label":"Reply to this email","type":"1"},{"id":"prompt_reply_advanced","label":"Reply to this thread","type":"1"},{"id":"prompt_summarize_this","label":"Summarize this","type":"0"},{"id":"prompt_translate_this","label":"Translate this","type":"0"}] [mzta-logger.js:35:44](moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/mzta-logger.js)
[ThunderAI Logger | mzta-popup] tabType: mail [mzta-logger.js:35:44](moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/mzta-logger.js)
[ThunderAI Logger | mzta-popup] filteredData: [{"id":"prompt_add_tags","label":"Add tags to this email","type":"1"},{"id":"prompt_classify","label":"Classify","type":"0"},{"id":"prompt_reply","label":"Reply to this email","type":"1"},{"id":"prompt_reply_advanced","label":"Reply to this thread","type":"1"},{"id":"prompt_summarize_this","label":"Summarize this","type":"0"},{"id":"prompt_translate_this","label":"Translate this","type":"0"}] [mzta-logger.js:35:44](moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/mzta-logger.js)
[ThunderAI Logger | mzta-background] Executing shortcut, promptId: prompt_add_tags [mzta-logger.js:35:44](moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/mzta-logger.js)
[ThunderAI Logger | mzta_Menus] fullPrompt: prompt_add_tags_full_text  "A lot has happened on Facebook since you last logged in. Here are some notifications you've missed from your friends. Yasha Iravantchi Notifications 4 new notifications Yasha Iravantchi You have 4 unread notifications to review A lot has happened on Facebook since you last logged in. Here are some notifications you've missed from your friends. NOTIFICATIONS 4 new notifications View your notifications Go to Facebook Was this email:Useful | Not Useful This message was sent to yiravan@umich.edu. If you don't want to receive these emails from Meta in the future, please unsubscribe. Meta Platforms, Inc., Attention: Community Support, 1 Meta Way, Menlo Park, CA 94025 To help keep your account secure, please don't forward this email. Learn more"  Limit the number of tags to 3. [mzta-logger.js:35:44](moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/mzta-logger.js)
>>>>>>>>>>>> [ThunderAI] tokensDone: Useful | Not Useful [mzta-add-tags.js:83:33](moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/special_commands/mzta-add-tags.js)
[ThunderAI Logger | mzta_Menus] mail_tags: Useful | Not Useful [mzta-logger.js:35:44](moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/mzta-logger.js)
>>>>>>>>>>>> full_tags_list: ["Important, Work, Personal, To Do, Later",{"$label1":{"tag":"Important","color":"#FF0000","key":"$label1","ordinal":""},"$label2":{"tag":"Work","color":"#FF9900","key":"$label2","ordinal":""},"$label3":{"tag":"Personal","color":"#009900","key":"$label3","ordinal":""},"$label4":{"tag":"To Do","color":"#3333FF","key":"$label4","ordinal":""},"$label5":{"tag":"Later","color":"#993399","key":"$label5","ordinal":""}}] [mzta-menus.js:233:33](moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/mzta-menus.js)
Error executing action for menu item prompt_add_tags: TypeError: curr_message is null
    act moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/mzta-menus.js:234
    executeMenuAction moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/mzta-menus.js:361
    <anonymous> moz-extension://d5050f32-e639-437b-a1f4-802510440484/mzta-background.js:222
[mzta-menus.js:364:25](moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/mzta-menus.js)
    executeMenuAction moz-extension://d5050f32-e639-437b-a1f4-802510440484/js/mzta-menus.js:364
    <anonymous> moz-extension://d5050f32-e639-437b-a1f4-802510440484/mzta-background.js:222
>>>>>>>>>>>>>>>>> response: false [mzta-popup.js:263:10](moz-extension://d5050f32-e639-437b-a1f4-802510440484/popup/mzta-popup.js)

**Comment by micz** [comment]
Thank you, I found the problem: the `fullPrompt` is not translated:

`[ThunderAI Logger | mzta_Menus] fullPrompt: prompt_add_tags_full_text [...]`

Instead of `prompt_add_tags_full_text` there should be the translated prompt. This is causing also the curr_message is null error, that has to be fixed, but happens only with a prompt without data placeholders.

I work on it.

**Comment by micz** [comment]
As a workaround you could just add a space in the add_tags prompt and save it.
Then it should work.

**Comment by micz** [comment]
Fixed in [3.0.0pre2](https://github.com/micz/ThunderAI/releases).
Thank you all.
